### ISSUE TITLE
Feature request:  Ability to show the pixel values along an arbitrary line-cut across the images

### ISSUE THREAD

**Issue by sanbw** [CARTAvis/carta-backend#1167]
This is a request for a feature where the pixel values along a line between two arbitrary points across the images are plotted as a line plot.  Currently CARTA displays line cuts along the x- and y-axis only, about the pixel where the mouse pointer is.  A similar plot, but along a an arbitrary line cut will be most useful.

Additionally, such display should optionally be displayed for *all* the appended images.  Currently the line plot is for only the image that is visible.

**Comment by kswang1029** [comment]
Thank you for the feature request. It is on our plan. In the _fall_ release (not the coming spring release), we will support at least a line cut region. Then we will extend a line to a polyline and an arbitrary curve.

**Comment by sanbw** [comment]
Thank you for the information.

**Comment by kswang1029** [comment]
spatial profile along a line or a polyline region is now implemented in v3-beta3 release and will be also in the v3-stable release in summer 2022.

**Comment by sanbee** [comment]
Thanks @kswang1029.  I tried it with real-valued images, it seems to work as you describe able.  However for complex-valued images the profile along a line (or an arbitrary track) isn't displayed.

**Comment by kswang1029** [comment]
> Thanks @kswang1029. I tried it with real-valued images, it seems to work as you describe able. However for complex-valued images the profile along a line (or an arbitrary track) isn't displayed.

Odd, based on my test, line and polyline spatial profile from a complex image should work. What's the image size in pixels? You may need to wait a bit if the line cut is long. 

https://user-images.githubusercontent.com/20819712/172186749-e52467cc-3496-4613-99c4-9dea33be54ab.mov

**Comment by sanbee** [comment]
It's a 2K x 2K complex image.  
The error message (copied below) from the CARTA server is perhaps useful.  The image coordinates are in the spatial frequency domain (UV-domain).
```
[CARTA] [error] Line region 1 spatial profile  failed: Cannot approximate line with no direction coordinate.
```
![snapshot1](https://user-images.githubusercontent.com/11493206/172204490-15e1944d-8fef-44c2-8ed8-34e3c64e0813.png)

**Comment by kswang1029** [comment]
@sanbee thanks for the info. The reason why it does not work is at the moment CARTA assumes the image needs to have valid direction coordinates so that CARTA can firstly figure out if the image is flat or not (projection distortion) then work out the sampling boxes along the line or the polyline trajectory. For this UV coordinate, we did not consider the case. I will file an enhancement issue for this (https://github.com/CARTAvis/carta-backend/issues/1121). Thanks for reporting.

**Comment by sanbee** [comment]
Thanks @kswang1029.  

This looks more like a bug to me than enhancement since CARTA does support non-direction coordinates in other places.

**Comment by veggiesaurus** [comment]
> Thanks @kswang1029.  
> 
> This looks more like a bug to me than enhancement since CARTA does support non-direction coordinates in other places.

Agreed. We should support fallback to pixel coordinates like we do for other features.

**Comment by kswang1029** [comment]
https://github.com/CARTAvis/carta-backend/issues/1311 is created to proceed.

profile along arbitrary trajectory still needs to be implemented.
